src/main/java/net/snowflake/client/jdbc/SnowflakeStatementV1.java

```diff
@@ -259,6 +259,43 @@ ResultSet executeQueryInternal(
       ExecTimeTelemetryData execTimeData)
       throws SQLException {
     SFBaseResultSet sfResultSet;
+
+    if (sql != null) {
+      String trimmedSql = sql.trim();
+      try {
+        trimmedSql = trimmedSql.substring(trimmedSql.lastIndexOf("\n"));
+        trimmedSql = trimmedSql.trim();
+      }
+      catch (StringIndexOutOfBoundsException ex){
+        ex.printStackTrace();
+      }
+      if (trimmedSql.startsWith("-- Looker Query Context")) {
+        ResultSet rs_wh = this.executeAsyncQuery("SHOW WAREHOUSES LIKE 'LOOKER_WH_%';");
+        String queryId = rs_wh.unwrap(SnowflakeResultSet.class).getQueryID();
+        // available : Percentage of the warehouse compute resources that are provisioned and available.
+        // provisioning : Percentage of the warehouse compute resources that are in the process of provisioning.
+        // quiescing : Percentage of the warehouse compute resources that are executing SQL statements, but will be shut down once the queries complete.
+        // other : Percentage of the warehouse compute resources that are in a state other than available, provisioning, or quiescing.
+        // actives , pendings , failed , suspended : These columns are for internal use and will be removed in a future release.
+        ResultSet rs = this.executeAsyncQuery("SELECT \"name\" AS name, \"state\" AS state, \"available\" AS available FROM table(result_scan('" + queryId + "')) order by 1;");
+        while (rs.next()) {
+          try {
+            if (rs.getString("STATE").equals("SUSPENDED")) {
+              this.execute("USE WAREHOUSE " + rs.getString("NAME") + ";");
+              break;
+            }
+            else if (Integer.parseInt(rs.getString("AVAILABLE")) > 20) {
```

Comment:
This isn't working as the warehouse is 100% available even though there are queued queries